tests/test_potentials.py

```diff
@@ -11,24 +11,42 @@ def harmonic_bond():
     return potentials.HarmonicBond(bond_idxs).bind(params)
 
 
-@pytest.fixture
-def summed_potential(harmonic_bond):
-    return potentials.SummedPotential([harmonic_bond], [harmonic_bond.params]).bind([harmonic_bond.params])
+def execute_bound_impl(bp):
+    coords = np.zeros(shape=(3, 3), dtype=np.float32)
+    box = np.diag(np.ones(3))
+    lam = 1
+    bp.execute(coords, box, lam)
 
 
-def test_summed_potential_raises_on_inconsistent_lengths(harmonic_bond):
+def test_bound_potential_keeps_referenced_potential_alive(harmonic_bond):
+    bp = harmonic_bond.bound_impl(np.float32)
+
+    # segfaults if referenced potential has been deallocated prematurely
+    execute_bound_impl(bp)
```

Comment:
Also checked that this does indeed segfault if the `smart_ptr` reference to the underlying potential in `BoundPotential` is replaced with a bare pointer (3515563)/